tendermint/src/abci/types.rs

```diff
@@ -29,6 +29,10 @@ pub struct VoteInfo {
     pub validator: Validator,
     /// Whether or not the validator signed the last block.
     pub sig_info: BlockSignatureInfo,
+    /// Non-deterministic extension provided by the sending validator's application.
+    pub vote_extension: Bytes,
+    /// Signature for the vote extension.
+    pub extension_signature: Option<Signature>,
```

Comment:
Should probably explicitly state that these are only relevant to v0.38+.

Reply:
https://github.com/informalsystems/tendermint-rs/pull/1312/commits/6a19f6c3b18ef9d7c24e89456eb6a4e2fe1a8879